blog/2019-02-19-reliable-microservices-data-exchange-with-the-outbox-pattern.adoc

```diff
@@ -0,0 +1,598 @@
+= Reliable Microservices Data Exchange With the Outbox Pattern
+gmorling
+:awestruct-tags: [ discussion, examples ]
+:awestruct-layout: blog-post
+
+If you've built a couple of microservices,
+you'll probably agree that the https://blog.christianposta.com/microservices/the-hardest-part-about-microservices-data/[hardest part about them is data]:
+microservices don't exist in isolation and very often they need to propagate data and data changes amongst each other.
+
+For instance consider a microservice that manages purchase orders:
+when a new order is placed, information about that order may have to be relayed to a shipment service
+(so it can assemble shipments of one or more orders) and a customer service
+(so it can update things like the customer's total credit balance based on the new order).
+
+There are different approaches for letting the order service know the other two about new purchase orders;
+e.g. it could invoke some https://en.wikipedia.org/wiki/Representational_state_transfer[REST], https://grpc.io/[grpc] or other (synchronous) API provided by these services.
+This might create some undesired coupling, though: the sending service must know which other services to invoke,
+and it also must be prepared for these services temporarily not being available.
+Service meshes such as https://istio.io/[Istio] come in helpful here, by providing capabilities like circuit breakers,
+request routing and much more.
+
+One general issue of such a synchronous approach is that it lacks re-playability,
+i.e. the possibility for new consumers to arrive after events have been sent and still be able to consume the entire event stream from the beginning.
+
+This can be achieved by using an asynchronous data exchange approach:
```

Comment:
I think this is not fully correct - this implies that `asynchronous data exchange` == `replayabilty` but those two concepts overlap only partially.